enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+    EIPAllocations []configv1.EIPAllocations `json:"eipAllocations"`
+}
+
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+#### 1. Set EIP through installer for the default ingress controller:
+   
+   a. The admin sets the `eipAllocations` in the installer using the install-config.yaml.
+   
+   b. The installer then sets the ingress cluster object with the `eipAllocations`.
+   
+   c. The cluster ingress operator then creates a default ingress controller by:
+
+     i.   creating an default ingress controller CR
+     
+     ii.  then scaling a deployment for the Ingress Controller CR which has the value of the `eipAllocations` from the
+          ingress cluster object.
+     
+     iii. then creating a service of service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`,
+          which uses the value from the field `eipAllocations` of ingress controller CR.
+
+#### 2. Set EIP through the Ingress Controller CR:
+   
+   The admin sets the `eipAllocations` in the Ingress Controller CR. Then same process
+   as per point 1.c.ii and 1.c.iii is followed by the ingress operator. 
+
+#### 3. Update the existing EIP for the IngressController:
+   
+   When the admin changes the eipAllocations field in the Ingress Controller CR, the ingress operator sets the Load Balancer
+   Progressing status as per [section](#Cluster administrator updates eipAllocations field of the existing ingress controller)
+   Then step 1.c.iii is followed by the ingress operator.
+
+   If the admin wants to detect the change in the eipAllocations and update the LB service automatically.
+   rather than deleting the Service explicitly, it is possible to annotate the IngressController with
+   the newly defined ingress.operator.openshift.io/auto-delete-load-balancer annotation. 
+   If the operator observes that this annotation is set and that its 
+   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.eipAllocations` field has been changed, 
+   the operator automatically deletes the Service if needed to complete an eip change operation. 
+   This purpose of this annotation is to simplify automation: A tool can set the annotation when it 
+   creates the IngressController or when it updates `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.eipAllocations` 
+   to instruct the operator to perform the operation automatically. This annotation is not intended for
+   end-users to use directly.
```

Comment:
You've already explained this earlier, I don't think it need to be lengthily explained again here.

```suggestion
   If the admin wants to detect the change in `eipAllocations` and update the LB service automatically, 
   rather than explicitly deleting the Service after changes, they can add the  ingress.operator.openshift.io/auto-delete-load-balancer annotation. 
```

Reply:
OK